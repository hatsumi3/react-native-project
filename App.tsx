import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  View,
  useColorScheme,
} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';

import Multi from './src/components/Multi';

function App(): JSX.Element {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const isNumber = 0;

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor="#0000ff"
        hidden={false}
        translucent={false}
      />
      <View>
        <Text style={styles.text}>My First component.</Text>
        <Text>{7 + 19 + 16}</Text>
        {false}
        {null}
        {undefined}
        {isNumber ? <Text>42</Text> : <Text>foo</Text>}
        {true && <Text>true</Text>}
        <Multi />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f5fcff',
  },
  text: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

export default App;
