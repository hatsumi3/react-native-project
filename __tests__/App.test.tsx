/**
 * @format
 */

import 'react-native';

import React from 'react';
import App from '../App';

// Note: import explicitly to use the types shiped with jest.
import {it, expect} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {ReactTestRenderer} from 'react-test-renderer';

jest.useFakeTimers();

it('renders correctly', async () => {
  let tree: ReactTestRenderer | undefined;

  await renderer.act(async () => {
    tree = renderer.create(<App />);
  });

  if (tree) {
    expect(tree.toJSON()).toMatchSnapshot();
  } else {
    throw new Error('Failed to render the component');
  }
});
