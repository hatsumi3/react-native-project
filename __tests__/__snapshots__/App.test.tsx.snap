// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "backgroundColor": "pink",
            "flex": 2,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          My First component.
        </Text>
        <Text>
          42
        </Text>
        <Text>
          foo
        </Text>
        <Text>
          true
        </Text>
        <View>
          <Text
            onPress={[Function]}
            style={
              {
                "backgroundColor": "green",
              }
            }
          >
            default text
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          Text
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 20,
              },
              {
                "color": "red",
                "fontWeight": "bold",
              },
            ]
          }
        >
          Text
        </Text>
        <Image
          resizeMode="contain"
          source={
            {
              "uri": "kafka_icon",
            }
          }
          style={
            {
              "height": 200,
              "width": 400,
            }
          }
        />
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          value={false}
        />
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "black",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        />
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "color": "black",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              {
                "backgroundColor": "white",
                "color": "black",
              }
            }
          >
            切り替え
          </Text>
          <Text
            style={
              {
                "backgroundColor": "white",
                "color": "black",
              }
            }
          >
            journal
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "blue",
            "flex": 1,
          }
        }
      >
        <Text>
          box2
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "pink",
            "height": 400,
          }
        }
      >
        <RCTScrollView
          data={
            [
              {
                "id": "first",
                "title": "ひとつめ",
              },
              {
                "id": "second",
                "title": "ふたつめ",
              },
              {
                "id": "thrid",
                "title": "みっつめ",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={[]}
          viewabilityConfigCallbackPairs={[]}
        >
          <View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <Text
                style={
                  {
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                ひとつめ
              </Text>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <Text
                style={
                  {
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                ふたつめ
              </Text>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <Text
                style={
                  {
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                みっつめ
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          {
            "backgroundColor": "blue",
            "height": 400,
          }
        }
      >
        <Text>
          box4
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
